src/main/java/selenium/pages/LoginPageWithPF.java

```diff
@@ -0,0 +1,33 @@
+package selenium.pages;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+import selenium.Driver;
+import selenium.TestUser;
+
+public class LoginPageWithPF extends BasePage {
+    @FindBy(id = "user-name")
+    private WebElement UsernameInput;
+    @FindBy(id = "password")
+    private WebElement PasswordInput;
+    @FindBy (id = "login-button")
+    private WebElement LoginButton;
+
+    private String url = "https://www.saucedemo.com/";
+
+    public LoginPageWithPF () {
+        this.driver = Driver.getInstance();
+        PageFactory.initElements(this.driver, this);
```

Comment:
You can initialize web elements on a page in a constructor of your base class

Reply:
Done